jgrapht-core/src/test/java/org/jgrapht/graph/AsUnweightedGraphTest.java

```diff
@@ -79,6 +79,14 @@
         assertEquals(Graph.DEFAULT_EDGE_WEIGHT, this.unweightedGraph.getEdgeWeight(loop), 0);
     }
 
+    @Test public void setEdgeWeight()
+    {
+        this.unweightedGraph.setEdgeWeight(e23, 81);
```

Comment:
The contract on the setEdgeWeight method in the graph interface says that if the graph does not support weights (i.e. it's an unweighted graph), then setEdgeWeight should throw UnsupportedOperationException.